jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/ContainerBuildPlan.java

```diff
@@ -93,6 +94,38 @@ public Builder setOsHint(String osHint) {
       return this;
     }
 
+    /**
+     * Adds a desired platform i.e image os and architecture pair. If the base image reference is a
+     * Docker manifest list or an OCI image index, must be set so that an image builder can select
+     * the image matching the given platform. If the base image reference is an image manifest, this
+     * value is ignored and the platform of the built image follows that of the base image. The
+     * default is {@code linux amd64 }.
+     *
+     * @param os value to select a base image in case of a manifest list
+     * @param architecture value to select a base image in case of a manifest list
+     * @return this
+     */
+    public Builder addPlatform(String os, String architecture) {
+      this.platforms.add(new Platform(os, architecture));
+      return this;
+    }
+
+    /**
+     * Sets a desired platform list i.e os and architecture pairs. If the base image reference is a
+     * Docker manifest list or an OCI image index, must be set so that an image builder can select
+     * the images matching the given platforms. If the base image reference is an image manifest,
+     * this value is ignored and the platform of the built image follows that of the base image. The
+     * default is {@code linux amd64 }.
+     *
+     * @param platforms a list of platform objects to be used to select base images in case of a
+     *     manifest list
+     * @return this
+     */
+    public Builder setPlatforms(List<Platform> platforms) {
+      this.platforms = new ArrayList<>(platforms);
```

Comment:
Should we guard against passing an empty input?

```java
    if (platforms.isEmpty()) {
      throw new IllegalArgumentException("some error message telling the caller that the list cannot be empty");
    }
```

If we disallow an empty list, it should also be documented in the Javadoc.

Reply:
Good Point !! I think its necessary to disallow empty lists !! It ensures that we always have a value in platforms.

Reply:
In our case, we have set up a system for checking null, so unless marked with `@Nullable`, our code assumes it's never null. Of course, because this is a public API and our internal null checking system can't prevent the user from passing null, but I think it's fine. They should understand that our API doesn't accept null unless marked with `@Nullable`.

Reply:
You can also use `Preconditions.checkArgument()`